src/beta_inc.jl

```diff
@@ -0,0 +1,945 @@
+    w *= c/b
+
+    #COMBINING 
+    u = d*log1p(a/b)
+    v = a*(log(b) - 1.0)
+    if u <= v
+        return w - u - v
+    end
+    return w - v - u
+end 
+
+"""
+    bcorr(a0,b0)
+
+Compute del(a0) + del(b0) - del(a0 + b0) where del(a) is given by:
+```math
+log(\\Gamma(a)) = (a - 0.5)*log(a) - a + 0.5*log(2\\pi) + del(a)
+```
+for a0,b0 >= 8
+"""
+function bcorr(a0::Float64, b0::Float64)
+    a = min(a0,b0)
+    b = max(a0,b0)
+
+    h = a/b
+    c = h/(1.0 + h)
+    x = 1.0/(1.0 + h)
+    x² = x*x
+    #SET SN = (1-X^N)/(1-X)
+    s₃ = 1.0 + (x + x²)
+    s₅ = 1.0 + (x + x²*s₃)
+    s₇ = 1.0 + (x + x²*s₅)
+    s₉ = 1.0 + (x + x²*s₇)
+    s₁₁ = 1.0 + (x + x²*s₉)
+    t = (1.0/b)^2.0
+    w = @horner(t, .833333333333333E-01, -.277777777760991E-02*s₃, .793650666825390E-03*s₅, -.595202931351870E-03*s₇, .837308034031215E-03*s₉, -.165322962780713E-02*s₁₁)
+    w *= c/b
+    # COMPUTE DEL(A) + W
+    t = (1.0/a)^2.0
+    return @horner(t, .833333333333333E-01, -.277777777760991E-02, .793650666825390E-03, -.595202931351870E-03, .837308034031215E-03, -.165322962780713E-02)/a + w
+end
+
+"""
+    esum(mu,x)
+
+Compute ``e^{mu+x}``
+"""
+function esum(mu::Float64, x::Float64)
+    if x > 0.0
+        if mu > 0.0 || mu + x < 0.0
+            w = mu
+            return exp(w)*exp(x)
+        else
+            return exp(mu + x)
+        end
+    elseif mu < 0.0 || mu + x > 0.0
+        w = mu
+        return exp(w)*exp(x)
+    else
+        return exp(mu + x)
+    end
+end
+
+"""
+    brcmp1(mu,a,b,x,y,case)
+
+If case == 1: compute ``e^{mu} * x^{a}y^{b}/B(a,b)``
+else : compute above with mu = 0.0
+"""
+function brcmp1(mu::Float64, a::Float64, b::Float64, x::Float64, y::Float64, case::Bool)
+    a0 = min(a,b)
+    if a0 >= 8.0
+      @goto l100
+    elseif x > 0.375
+      @goto l10
+    end
+    lnx = log(x)
+    lny = log1p(-x)
+    @goto l20
+
+    @label l10
+     if y > 0.375
+      @goto l11
+     end
+     lnx  = log1p(-y)
+     lny = log(y)
+     @goto l20
+   
+    @label l11
+     lnx = log(x)
+     lny = log(y)
+    @label l20
+     z = a*lnx + b*lny
+     if a0 < 1.0
+       @goto l30
+     end
+     z -= logbeta(a,b)
+     return case ? esum(mu,z) : exp(z)
+
+   #PROCEDURE A < 1 OR B < 1
+    @label l30
+     b0 = max(a,b)
+     if b0 >= 8.0
+        @goto l80
+     elseif b0 > 1.0
+        @goto l60
+     end
+
+     # FOR B0 <= 1
+     ans = case ? esum(mu,z) : exp(z)
+     if ans == 0.0
+        return 0.0
+     end
+     apb = a + b
+     if apb > 1.0
+        @goto l40
+     else
+        z = 1.0 + SpecialFunctions.rgamma1pm1(apb)
+        @goto l50
+     end
+    @label l40
+     u = a + b - 1.0
+     z = (1.0 + SpecialFunctions.rgamma1pm1(u))/apb
+
+    @label l50
+     c = (1.0 + SpecialFunctions.rgamma1pm1(a))*(1.0 + SpecialFunctions.rgamma1pm1(b))/z
+     return ans*(a0*c)/(1.0 + a0/b0)
+
+    #  FOR 1 < B0 < 8
+    @label l60
+     u = SpecialFunctions.loggamma1p(a0)
+     n = b0 - 1.0
+     if n < 1
+        @goto l70
```

Comment:
rewrite this to avoid the `@goto l70`:
```
if n >= 1
     c = 1.0
     for i = 1:n
        b0 -= 1.0
        c *= (b0/(a0+b0))
     end
     u += log(c)
end
```